datumaro/plugins/coco_format/extractor.py

```diff
@@ -236,19 +274,37 @@ def __call__(self):
             return mask_utils.merge(rles)
 
     def _get_label_id(self, ann):
-        if not ann["category_id"]:
+        cat_id = self._parse_field(ann, "category_id", int)
+        if not cat_id:
             return None
-        return self._label_map[ann["category_id"]]
+
+        label_id = self._label_map.get(cat_id)
+        if label_id is None:
+            raise InvalidLabelError(str(cat_id))
+        return label_id
+
+    def _parse_field(
+        self, ann: Dict[str, Any], key: str, cls: Union[None, Type[T], Tuple[Type, ...]]
+    ) -> T:
```

Comment:
I don't think that's a well-defined type annotation. If the user passes `None` or a tuple of types, then there's no way for a type checker to derive `T`.

Also, `None` seems unnecessary here. I don't think it's used anywhere, and if it ever needs to be used, the same effect can be accomplished by passing `object`.

Reply:
Fixed

Reply:
Added overloads